dashboard/test/controllers/activities_controller_test.rb

```diff
@@ -628,69 +600,6 @@ def expect_controller_logs_milestone_regexp(regexp)
     end
   end
 
-  test "anonymous milestone starting with empty session saves progress in section" do
-    sign_out @user
-
-    # do all the logging
-    @controller.expects :log_milestone
-    @controller.expects :slog
-
-    @controller.expects(:trophy_check).never # no trophy if not logged in
-
-    assert_creates(LevelSource) do
-      assert_does_not_create(Activity, UserLevel) do
-        post :milestone, @milestone_params.merge(user_id: 0)
-      end
-    end
-
-    # record activity in session
-    assert_equal 100, client_state.level_progress(@script_level.level_id)
-
-    # record the total lines of code in session
-    assert_equal 20, client_state.lines
-
-    assert_response :success
-
-    expected_response = build_expected_response(
-        total_lines: 20,
-        level_source: "http://test.host/c/#{assigns(:level_source).id}")
-    assert_equal_expected_keys expected_response, JSON.parse(@response.body)
-  end
-
-  test "anonymous milestone with existing session adds progress in session" do
-    sign_out @user
-
-    # set up existing session
-    client_state.set_level_progress(@script_level_prev.level_id, 50)
-    client_state.add_lines(10)
-
-    # do all the logging
-    @controller.expects :log_milestone
-    @controller.expects :slog
-
-    @controller.expects(:trophy_check).never # no trophy if not logged in
-
-    assert_creates(LevelSource) do
-      assert_does_not_create(Activity, UserLevel) do
-        post :milestone, @milestone_params.merge(user_id: 0)
-      end
-    end
-
-    # record activity in session
-    assert_equal 50, client_state.level_progress(@script_level_prev.level_id)
```

Comment:
Likewise we should keep this test and only get rid of the assertions about server side updates of client state.